src/scripts/main.js

```diff
@@ -1 +1,65 @@
 'use strict';
+
+const successEl = (message) =>
+  `<div data-qa="notification" class="success">${message}</div>`;
+const errorEl = (message) =>
+  `<div data-qa="notification" class="error">${message}</div>`;
+
+const displayMessage = (message) => {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('beforeend', message);
+};
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(errorEl('First promise was rejected'));
+  }, 3000);
+
+  const handleClick = () => {
+    clearTimeout(timeoutId);
+    resolve(successEl('First promise was resolved'));
+    document.removeEventListener('click', handleClick);
+  };
+
+  document.addEventListener('click', handleClick);
+});
+
+const promise2 = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve(successEl('Second promise was resolved'));
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleClick);
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const promise3 = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    } else if (e.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve(successEl('Third promise was resolved'));
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('contextmenu', handleClick);
+    }
```

Comment:
In your current implementation, the third promise will resolve when both left and right clicks have been made, regardless of the order. If you want to ensure that the left click is made before the right click, you can modify your logic to track the order of the clicks.